master/buildbot/process/buildstep.py

```diff
@@ -438,6 +438,8 @@ def startStep(self, remote):
             name=util.ascii2unicode(self.name))
         yield self.master.data.updates.startStep(self.stepid)
 
+        self.locks = yield self.build.render(self.locks)
```

Comment:
Can you please add a test where the locks are *not* renderable? I'm not sure this line will work in that case.